### PR TITLE
Fast App

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

An attempt to create an extra fast and performant variant of the native app. It's got tons of changes to reduce render size, network calls, animations, CLS, etc. While a lot of this code is not production ready, i think there are some good takeaways we can incorporate! Going to push this to the RC-Staging apps so we can try it out on the device, and compare to Staging

### Dragons

Major missing piece is collectibles. Im looking to bring it back, but i ran out of time today. Ideally we delay requesting and storing this data till someone goes to collectibles tab, but seems like the "has-collectibles" variable is not reliable. anyway will check with some folks tomorrow to work on.

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/fast-app

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/fast-app

File: packages/mobile/android/app/proguard-rules.pro [hunk 1]
```diff
@@ -12,4 +12,12 @@
 -keep class com.facebook.hermes.unicode.** { *; }
 -keep class com.facebook.jni.** { *; }
 -keep class com.swmansion.reanimated.** { *; }
--keep class com.facebook.react.turbomodule.** { *; }
\ No newline at end of file
+-keep class com.facebook.react.turbomodule.** { *; }
+-keep public class com.dylanvann.fastimage.* {*;}
```
Comment (sliptype) [inline packages/mobile/android/app/proguard-rules.pro:16]: Dylan collusion

File: packages/mobile/src/components/image/FastImage.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+} from 'react-native-fast-image'
+import RNFastImage from 'react-native-fast-image'
+
+export type FastImageProps = Partial<Omit<RNFastImageProps, 'source'>> & {
+  source?: ImageSourcePropType
+  priority?: Priority
+  size?: 'small' | 'medium' | 'large'
+}
+
+const sizeMap = {
```
Comment (sliptype) [inline packages/mobile/src/components/image/FastImage.tsx:14]: Ah ok this is what you meant. I like this as a temp solution! but long term we shouldn't be generating all the extra sources, my pr makes that change

File: packages/mobile/src/components/image/tmp.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
+export {}
```
Comment (sliptype) [inline packages/mobile/src/components/image/tmp.ts:1]: What's this?

File: packages/mobile/src/components/lineup/Lineup.tsx [hunk 4]
```diff
@@ -323,17 +326,17 @@ export const Lineup = ({
   // When scrolled past the end threshold of the lineup and the lineup is not loading,
   // trigger another load
   useEffect(() => {
-    if (isPastLoadThreshold && status !== Status.LOADING) {
+    if (isPastLoadThreshold && status !== Status.LOADING && inView) {
```
Comment (sliptype) [inline packages/mobile/src/components/lineup/Lineup.tsx:329]: Clutch

File: packages/mobile/src/components/skeleton/Skeleton.tsx [hunk 5]
```diff
@@ -35,33 +33,11 @@ const useStyles = makeStyles(({ palette }) => ({
...
-          useNativeDriver: true,
-          easing: Easing.ease
-        })
-      ),
-    [shimmerPos]
-  )
-
   return (
     <View style={[styles.view, { height, width }, style]}>
-      <Animated.View
```
Comment (sliptype) [inline packages/mobile/src/components/skeleton/Skeleton.tsx:56]: So removing the skeleton anim entirely here?

File: packages/mobile/src/components/top-tab-bar/CollapsibleTopTabNavigator.tsx [hunk 6]
```diff
@@ -91,7 +91,14 @@ export const CollapsibleTabNavigator = ({
       collapsibleOptions={collapsibleOptions}
       initialRouteName={initialScreenName}
       tabBar={tabBar}
-      screenOptions={screenOptions}
+      // lazy={true}
```
Comment (sliptype) [inline packages/mobile/src/components/top-tab-bar/CollapsibleTopTabNavigator.tsx:94]: Not using the lazy option?

File: packages/mobile/src/screens/collection-screen/CollectionScreenDetailsTile.tsx [hunk 7]
```diff
@@ -267,7 +272,7 @@ export const CollectionScreenDetailsTile = ({
       renderBottomContent={renderTrackList}
       headerText={!isOfflineModeEnabled ? headerText : undefined}
       renderHeader={isOfflineModeEnabled ? renderHeader : undefined}
-      renderImage={renderCustomImage ?? renderImage}
+      renderImage={renderImage ?? renderImage}
```
Comment (sliptype) [inline packages/mobile/src/screens/collection-screen/CollectionScreenDetailsTile.tsx:275]: Idk about this line 😂 

But the custom image render is used for smart collections, like Heavy Rotation and things like that

File: packages/mobile/src/screens/profile-screen/TracksTab.tsx [hunk 8]
```diff
@@ -31,45 +30,24 @@ export const TracksTab = () => {
...
 
-  if (!lineup) return null
-
-  /**
-   * If the profile isn't loaded yet, pass the lineup an empty entries
-   * array so only skeletons are displayed
-   */
   return (
     <Lineup
+      selfLoad
```
Comment (sliptype) [inline packages/mobile/src/screens/profile-screen/TracksTab.tsx:50]: Nice

File: packages/mobile/src/screens/user-list-screen/UserList.tsx [hunk 9]
```diff
@@ -97,14 +97,15 @@ export const UserList = (props: UserListProps) => {
     if (!isEmpty && !isRefreshing && !loading && isFocused) {
       cachedUsers.current = users
     }
-  }, [isEmpty, isRefreshing, isFocused, loading, users])
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (sliptype) [inline packages/mobile/src/screens/user-list-screen/UserList.tsx:100]: Maybe comment about why excluding `users`

**Review by sliptype — COMMENTED** [review]
HUGE lift, not approving yet bc I don't think we're planning on merging as is

**Comment by dylanjeffers** [comment]
> HUGE lift, not approving yet bc I don't think we're planning on merging as is

Totally!

File: packages/mobile/src/components/image/FastImage.tsx (hunk 2 above)
Comment (dylanjeffers) [reply packages/mobile/src/components/image/FastImage.tsx:14]: Ahh gotcha!

File: packages/mobile/src/components/image/tmp.ts (hunk 3 above)
Comment (dylanjeffers) [reply packages/mobile/src/components/image/tmp.ts:1]: No need anymore thanks

File: packages/mobile/src/components/skeleton/Skeleton.tsx (hunk 5 above)
Comment (dylanjeffers) [reply packages/mobile/src/components/skeleton/Skeleton.tsx:56]: Yeah not ling term solution, just honestly curious what it feels like without shimmer. Prob will replace with that skeleton lib

File: packages/mobile/src/components/top-tab-bar/CollapsibleTopTabNavigator.tsx (hunk 6 above)
Comment (dylanjeffers) [reply packages/mobile/src/components/top-tab-bar/CollapsibleTopTabNavigator.tsx:94]: Sadly no, lazy breaks on collapsible. But the alternative is the "inView" option in lineups to still wait before making request

File: packages/mobile/src/screens/collection-screen/CollectionScreenDetailsTile.tsx (hunk 7 above)
Comment (dylanjeffers) [reply packages/mobile/src/screens/collection-screen/CollectionScreenDetailsTile.tsx:275]: Omg that should be renderImageProo

File: packages/mobile/src/screens/user-list-screen/UserList.tsx (hunk 9 above)
Comment (dylanjeffers) [reply packages/mobile/src/screens/user-list-screen/UserList.tsx:100]: Totes

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/fast-app

**Comment by dylanjeffers** [comment]
All merged in with separate prs

**Pull request closed without merging**